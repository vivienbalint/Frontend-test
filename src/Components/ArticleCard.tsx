// import React from "react";
import { Link } from "react-router-dom";
import { Card } from "react-bootstrap";

type CardProps = {
  title: string;
  description: string;
  body: string;
  slug: string;
  isDisabled: boolean;
};

export default function ArticleCard(props: CardProps) {
  return (
    <Card border="dark" className="card mb-4">
      <Card.Body>
        <Card.Title>{props.title}</Card.Title>
        <Card.Subtitle className="mb-2 text-muted">
          {props.description}
        </Card.Subtitle>
        <Card.Text>{props.body}</Card.Text>
        {!props.isDisabled && (
          <Link to={`/article/${props.slug}`} className="card-link">
            <p>Read the whole Article </p>
          </Link>
        )}
      </Card.Body>
    </Card>
  );
}
