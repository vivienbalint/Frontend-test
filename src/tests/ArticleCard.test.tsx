import React from "react";
import { mount } from "enzyme";
import ArticleCard from "../Components/ArticleCard";
import "../setupTests";

describe("Article Card with isDisabled = true prop", () => {
  const propTitle = "An article";
  const propDesc = "A description";
  const propBody = "A long text";
  const propSlug = "A slug";
  const propIsDisabled = true;

  const wrapper = mount(
    <ArticleCard
      title={propTitle}
      description={propDesc}
      body={propBody}
      slug={propSlug}
      isDisabled={propIsDisabled}
    />
  );

  it("should accept propTitle", () => {
    expect(wrapper.props().title).toEqual(propTitle);
  });

  it("should accept propDesc", () => {
    expect(wrapper.props().description).toEqual(propDesc);
  });

  it("should accept propBody", () => {
    expect(wrapper.props().body).toEqual(propBody);
  });

  it("should accept propSlug", () => {
    expect(wrapper.props().slug).toEqual(propSlug);
  });

  it("should accept propIsDisabled", () => {
    expect(wrapper.props().isDisabled).toEqual(propIsDisabled);
  });

  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should not have a link", () => {
    expect(wrapper.find("a").length).toEqual(0);
  });
});
