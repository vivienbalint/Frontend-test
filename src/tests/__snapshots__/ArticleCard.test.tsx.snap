// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Card with isDisabled = true prop should match the snapshot 1`] = `"<div class=\\"card mb-4 card border-dark\\"><div class=\\"card-body\\"><div class=\\"card-title h5\\">An article</div><div class=\\"mb-2 text-muted card-subtitle h6\\">A description</div><p class=\\"card-text\\">A long text</p></div></div>"`;
